### ISSUE TITLE
Adding support for more math characters

### ISSUE THREAD

**Issue by oddhack** [asciidoctor/asciidoctor-pdf#1832]
Looking at some closed issues, I think there is some willingness/ability to add additional characters to the default font for e.g. math support. If so, we'd find all of the following very useful for the Vulkan docs:

:harr: ↔
:cdot: ⋅
:downarrow: ↓
:elem: ∈
:land: ∧
:lor: ∨
:oplus: ⊕
:lceil: ⌈
:rceil: ⌉
:lfloor: ⌊
:rfloor: ⌋
:forall: ∀
:check: ✓

**Comment by mojavelinux** [comment]
Are you expecting these in the primary font, or in the fallback font?

```
asciidoctor-pdf -a pdf-theme=default-with-fallback-font glyphs.adoc
```

The challenge with adding glyphs to the primary font is that we can only add characters that the font itself supports (in this case Noto Serif). (When I update the font, I'm not adding glyphs. I'm only not taking them away). So we have to check if it even has them.

The fallback font is missing glyphs for the *floor characters, but seems to have the rest.

**Comment by oddhack** [comment]
Symbols in an inconsistent font is far better than empty boxes, so having them in the fallback font sounds like a solution that would accomplish what is needed without requiring you to act as a font designer. The lack of ceiling/floor symbols is significant though. ISTM they're present in M+ 1mn (per https://www.fontspace.com/m-1mn-font-f13602) and M+ 1mn is included in the default-theme.yml font: for 1.5.3 along with Noto Serif, so I'm not clear on why they're not rendering (unless your M+ 1mn also does not include these symbols?)

**Comment by mojavelinux** [comment]
Right, I'm saying that M+ includes them in the full font, but we're losing the *floor glyphs when preparing the font for Asciidoctor PDF. So I can restore them. (We don't bundle the whole font because its enormous and makes the gem size explode).

I can definitely add these missing glyphs to M+ 1p (the fallback font), but I could also consider adding them in the monospaced font too (since it comes from the same family and may have them).

Do you want to use them in monospaced format or in regular format?

p.s. Keep in mind that you always have the option to use your own fallback font. The theming system allows you to extend the default theme to add an extra font with just a few lines of configuration.

**Comment by mojavelinux** [comment]
FYI here's the script that subsets the fonts for inclusion in the gem to give you can get a sense of how this process works: https://github.com/asciidoctor/asciidoctor-pdf/blob/master/scripts/subset-fonts.pe

**Comment by oddhack** [comment]
For our purposes the regular format is superior. I can see how to setup my own fallback font but then I'd be doing basically what you're talking about, except with far less knowledge of how to do it. Hopefully these symbols will be more generally useful than just the docs I edit (several other Khronos specs are also going to need them, at least).

**Comment by oddhack** [comment]
Oh, I think both floor and ceil are missing from the fallback - not just floor.

**Comment by mojavelinux** [comment]
> Oh, I think both floor and ceil are missing from the fallback - not just floor.

I can confirm that. I meant to say floor and ceil.

**Comment by oddhack** [comment]
> FYI here's the script that subsets the fonts for inclusion in the gem to give you can get a sense of how this process works: https://github.com/asciidoctor/asciidoctor-pdf/blob/master/scripts/subset-fonts.pe

So trying to parse this without actually understanding the scripting language, to add characters to the M+ fallback font ("code_font"?), would the right thing to do be add SelectMore() statements for the relevant glyphs inside the "if script == "subset")" block?.

**Comment by mojavelinux** [comment]
That is absolutely correct. The way the script works is that it starts with no selections, then it selects glyphs to include. Then it saves the font.

**Comment by oddhack** [comment]
> That is absolutely correct. The way the script works is that it starts with no selections, then it selects glyphs to include. Then it saves the font.

Thanks. Should I be working off the latest tag in the repo for purposes of generating our own font? I get a fatal error from the subset script when running from HEAD due to a missing (or more likely, misnamed) font.

**Comment by mojavelinux** [comment]
Resolved by commit 9432cb04767fdc47414cc41b4fef929efa8b1a94.
